matrix_client/room.py

```diff
@@ -655,6 +655,9 @@ def _process_state_event(self, state_event):
                 self.invite_only = econtent["join_rule"] == "invite"
             elif etype == "m.room.guest_access":
                 self.guest_access = econtent["guest_access"] == "can_join"
+            elif etype == "m.room.encryption":
+                if econtent.get("algorithm") == "m.megolm.v1.aes-sha2":
```

Comment:
According to the e2e implementation guide, we also need to handle events with algorithm = `m.olm.v1.curve25519-aes-sha2`. Not sure if that's true though, since the section right above about enabling e2e says that only `m.megolm.v1.aes-sha2` is permitted. Could you clarify this please?

Reply:
Hm, i'm surprised the guide says you need to implement both. I'm not aware of any legitimate use for pure olm encryption *within a room* currently, other than compatibility with ~2016 vintage Matrix E2E.  Pure olm is of course used for toDevice messages however.

Reply:
@non-Jedi I can't find where this would be written in the guide, maybe you confused `m.room.encryption` and `m.room.encrypted`?  
Olm encrypted events are indeed handled as toDevice events in later parts.

Reply:
Yep. @Zil0 got it. Exactly what happened. (wish github had a way to mark specific comment chains as resolved rather than just whole reviews...)